prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -513,14 +520,19 @@ class DetailedSubmissionHistory(
 
         if (destinations.size == 0) {
             /**
-             * Cases where this may hit:
-             *     1) Data hasn't been processed yet (common in async submissions)
-             *     2) Very rare: No data matches any geographical location.
-             *         e.g. If both the testing tab and patient data were foreign addresses.
-             * At the moment we have NO easy way to distinguish the latter rare case,
-             * so it will be treated as status RECEIVED as well.
+             * This conditional serves to differentiate where a report was submitted async and therefore hasn't
+             * been processed yet vs. a report that has been processed, but did not have any eligible receivers.
+             *
+             * The most likely scenario for that is when the item does not pass the jurisdictional filter for any of
+             * the receivers.
+             *
+             * Note: This method only works for the universal pipeline as the covid pipeline does the filtering and
+             * routing in one step.
              */
-            return Status.RECEIVED
+            if (actionsPerformed == setOf(TaskAction.receive)) {
```

Comment:
If you put a breakpoint here:
- a report going through the UP will initially have a receive action and then a route action even if there is ultimately no receivers
- the Covid pipeline will only ever have a receive action since there is no separate route step